cmd/util.go

```diff
@@ -32,280 +32,96 @@ func isInNamespace(meta metav1.ObjectMeta, namespace string) (valid bool) {
 	return namespace == apiv1.NamespaceAll || namespace == meta.Namespace
 }
 
-func convertDeploymentToDeploymentList(deployment Deployment) (deploymentList *DeploymentList) {
-	if isInRootConfigNamespace(deployment.ObjectMeta) {
-		deploymentList = &DeploymentList{Items: []Deployment{deployment}}
-	} else {
-		deploymentList = &DeploymentList{Items: []Deployment{}}
+func getContainers(resource k8sRuntime.Object) (container []Container) {
+	switch kubeType := resource.(type) {
+	case *StatefulSet:
+		container = kubeType.Spec.Template.Spec.Containers
+	case *DaemonSet:
+		container = kubeType.Spec.Template.Spec.Containers
+	case *ReplicationController:
+		container = kubeType.Spec.Template.Spec.Containers
+	case *Deployment:
+		container = kubeType.Spec.Template.Spec.Containers
```

Comment:
Since only pod is different than rest we can collapse all the case in the `default` behavior same thing can be done with other `switch` statements

Reply:
Unfortunately, that doesn't work because in a `default` you end up with an `runtime.Object` which doesn't have `Spec` only after type switching will the underlying methods and types be accessible.

```kubeType.Spec undefined (type "github.com/Shopify/kubeaudit/vendor/k8s.io/apimachinery/pkg/runtime".Object has no field or method Spec)```

I wish I had a more satisfying answer here for you. If you know of a way to make it work I will gladly do it that way.